lib/retry.js

```diff
@@ -0,0 +1,175 @@
+const assert = require('node:assert')
+
+const createError = require('http-errors')
```

Comment:
not a fan, this adds 6 dependencies

Reply:
Yeah, agree. I’d better create an util or something 👍